power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -709,27 +725,29 @@ class Deserializer {
             }
             return found->second;
         }();
+        auto const attribute_buffer_order = get_attribute_buffer_mapping(*buffer.buffer, attributes);
+        buffer.attribute_order = &attribute_buffer_order;
```

Comment:
```suggestion
        buffer.attribute_order = get_attribute_buffer_mapping(*buffer.buffer, attributes);
```

As suggested above, assign directly by value.